File: 2.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>
using namespace std;


bool isPrime(int num) {
    if (num <= 1) return false;
    if (num == 2) return true;
    if (num % 2 == 0) return false;  
    for (int i = 3; i <= sqrt(num); i += 2) {
        if (num % i == 0) return false;
    }
    return true;
}

vector<int> rearrangeArray(vector<int>& arr) {
    vector<int> primes;
    vector<int> nonPrimes;

    for (int num : arr) {
        if (isPrime(num)) {
            primes.push_back(num);
        } else {
            nonPrimes.push_back(num);
        }
    }

 
    sort(primes.begin(), primes.end(), greater<int>());
    sort(nonPrimes.begin(), nonPrimes.end(), greater<int>());

    vector<int> result;

   
    if (!primes.empty()) {
        result.push_back(primes.front());
        primes.erase(primes.begin());
    }

    
    result.insert(result.end(), nonPrimes.begin(), nonPrimes.end());

    if (!primes.empty()) {
        result.push_back(primes.back());
    }

    return result;
}

int main() {
    vector<int> arr1 = {1, 8, 11, 2, 5};  

    
    vector<int> result1 = rearrangeArray(arr1);

   
    cout << "Rearranged Array 1: ";
    for (int num : result1) {
        cout << num << " ";
    }
    cout << endl;

    return 0;
}

File: 2.java
import java.util.*;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Rearrange {
    
   
    public static boolean isPrime(int num) {
        if (num <= 1) return false;
        for (int i = 2; i * i <= num; i++) {
            if (num % i == 0) return false;
        }
        return true;
    }
    
    public static int[] rearrange(int[] arr) {
        List<Integer> primes = new ArrayList<>();
        List<Integer> nonPrimes = new ArrayList<>();
        
      
        for (int num : arr) {
            if (isPrime(num)) {
                primes.add(num);
            } else {
                nonPrimes.add(num);
            }
        }
        
      
        Collections.sort(primes, Collections.reverseOrder());
        
       
        Collections.sort(nonPrimes, Collections.reverseOrder());
        
     
        List<Integer> result = new ArrayList<>();
        
     
        result.add(primes.get(0));
        
     
        result.addAll(nonPrimes);
        
       
        result.add(primes.get(primes.size() - 1));
        

        return result.stream().mapToInt(i -> i).toArray();
    }

    public static void main(String[] args) {
      
        int[] arr1 = {5, 1, 8, 11, 2};
        int[] result1 = rearrange(arr1);
        System.out.println(java.util.Arrays.toString(result1));
      
        int[] arr2 = {1, 6, 4, 13, 9, 3};
        int[] result2 = rearrange(arr2);
        System.out.println(java.util.Arrays.toString(result2)); 
    }
}